### ISSUE TITLE
Suggestion: rename `IDMixin` to `ID`

### ISSUE THREAD

**Issue by apdavison** [SpiNNakerManchester/sPyNNaker#1041]
I noticed when debugging a script using spyNNaker that objects of type "IDMixin" appear where I would expect to see type "ID". The name "IDMixin" is not intended to be part of the PyNN API. In the PyNN "common" implementation, the class is used as a [mixin](https://en.wikipedia.org/wiki/Mixin) for subclassing int or long, it does not have an `__init__()`. I think the best solution would be to rename the class in spyNNaker to `ID`. This is of course not urgent, it has no functional effect, just caused me some confusion when debugging.

**Comment by Christian-B** [comment]
In PyNN if you do

pop_1 = sim.Population(10, ....
x = pop_1[2:3
y = pop_1[1]

x would be a PopulationView

What kind of Object should y be.

can y also be just a PopulationView too or does it need to be a different object and if so what?

**Comment by apdavison** [comment]
y should be an ID object, although this name is potentially confusing, and the class could be renamed Cell or Neuron.

**Comment by Christian-B** [comment]
"Cell" is already used for neuron model cell type and "neuron" is already over used.

I am thinking of using PopulationID  which goes nicely with PopulationView

**Comment by apdavison** [comment]
Using a different name from other PyNN implementations is confusing, that's why I opened this ticket. If we change, it should be done across all implementations. I've opened https://github.com/NeuralEnsemble/PyNN/issues/715 to discuss this question and the relationship to PopulationView. I'd appreciate input from the SpiNNaker side there.

**Comment by Christian-B** [comment]
I completely agree that  Using a different name from other PyNN implementations is confusing!

I will await the decision in  https://github.com/NeuralEnsemble/PyNN/issues/715 and then implement the same in SpyNNaker

**Comment by Christian-B** [comment]
Still awaiting the decision what to do here in https://github.com/NeuralEnsemble/PyNN/issues/715
